service/framework.go

```diff
@@ -0,0 +1,407 @@
+		c.Logger = config.Logger
+		c.ResourceRouter = framework.DefaultResourceRouter(resources)
+		c.TPR = newTPR
+
+		customObjectFramework, err = framework.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	return customObjectFramework, nil
+}
+
+func newPodFramework(config Config) (*framework.Framework, error) {
+	if config.Flag == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Flag must not be empty")
+	}
+	if config.Viper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Viper must not be empty")
+	}
+
+	var err error
+
+	var k8sClient kubernetes.Interface
+	{
+		c := k8sclient.DefaultConfig()
+
+		c.Address = config.Viper.GetString(config.Flag.Service.Kubernetes.Address)
+		c.Logger = config.Logger
+		c.InCluster = config.Viper.GetBool(config.Flag.Service.Kubernetes.InCluster)
+		c.TLS.CAFile = config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.CAFile)
+		c.TLS.CrtFile = config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.CrtFile)
+		c.TLS.KeyFile = config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.KeyFile)
+
+		k8sClient, err = k8sclient.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var podResource framework.Resource
+	{
+		c := podresource.DefaultConfig()
+
+		c.K8sClient = k8sClient
+		c.Logger = config.Logger
+
+		podResource, err = podresource.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resources []framework.Resource
+	{
+		resources = []framework.Resource{
+			podResource,
+		}
+
+		logWrapConfig := logresource.DefaultWrapConfig()
+
+		logWrapConfig.Logger = config.Logger
+
+		resources, err = logresource.Wrap(resources, logWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		retryWrapConfig := retryresource.DefaultWrapConfig()
+
+		retryWrapConfig.BackOffFactory = func() backoff.BackOff { return backoff.WithMaxTries(backoff.NewExponentialBackOff(), ResourceRetries) }
+		retryWrapConfig.Logger = config.Logger
+
+		resources, err = retryresource.Wrap(resources, retryWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		metricsWrapConfig := metricsresource.DefaultWrapConfig()
+
+		metricsWrapConfig.Name = config.Name
+
+		resources, err = metricsresource.Wrap(resources, metricsWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var newWatcherFactory informer.WatcherFactory
+	{
+		newWatcherFactory = func() (watch.Interface, error) {
+			options := apismetav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", key.PodWatcherLabel, config.Name),
+			}
+
+			watcher, err := k8sClient.CoreV1().Pods("").Watch(options)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			return watcher, nil
+		}
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.DefaultConfig()
+
+		c.WatcherFactory = newWatcherFactory
+
+		c.RateWait = 10 * time.Second
+		c.ResyncPeriod = 5 * time.Minute
```

Comment:
Why not using the defaults?

Reply:
No special reason. I think the values will change. I just do not know yet which values are better and why. Lets learn and adapt along the road.

Reply:
FYI: My thinking is it's nice to have all defaults inside operatorkit, and change it only when we have a particular reason for that. That gives us ability to change the default globally.